### PR TITLE
Add package tmate

### PR DESCRIPTION
**Pull request by DamienCassou**

### CODE REVIEW

File: pkgs/tools/misc/tmate/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  # It seems that the project requires cmake to be in the PATH, but
+  # the projet does not use cmake in the configure phase. The
+  # following disables the use of cmake for the configure phase.
+  configurePhase = ''
+    ./autogen.sh
+    configureFlags="--prefix=$prefix"
+    echo configure flags: $configureFlags
+    ./configure $configureFlags
+  '';
```
Comment (vcunat) [inline pkgs/tools/misc/tmate/default.nix]: Wouldn't it be better to just define `dontUseCmakeConfigure = true;`? (I didn't test it does exactly what you need.)

**Comment by DamienCassou** [comment]
@vcunat thanks, I fixed the pull request. With your comment, I finally understood how nixpkgs could have a different process with cmake even though cmake is not referenced in `setup.sh` :-). Thanks for that.

**Comment by jagajaga** [comment]
Please rename commit's name following our guide in `CONTRIBUTING.md`

**Comment by DamienCassou** [comment]
@jagajaga I changed the commit's comment. I didn't find any best-practice guide on naming branches. What do you mean by "rename commit's name"?

**Comment by vcunat** [comment]
He meant the commit message, I'm certain, which you fixed. What do you mean by "naming branches"?

**Comment by DamienCassou** [comment]
@vcunat the branch is named `new-tmate` and I was wondering if there was a standard naming scheme (e.g., `new/<package>`)

**Comment by jagajaga** [comment]
Branch name doesn't matter :)

**Pull request merged**